### ISSUE TITLE
AutoLoader should load only known core files

### ISSUE THREAD

**Issue by getdatakick** [thirtybees/thirtybees#948]
Thirtybees autoloader loads *every* php file in `classes` directory. Sometimes this can cause problems. I have seen following issues:

- merchant copied (by mistake in ftp manager) whole directory `classes/dir_a/*.php` to `classes/dir_b/dir_a/*.php`. Autoloader always loaded files from `classes/dir_b/dir_a`, and any updates to files in `classes/dir_a` was ignored
- there can be remnants of prestashop php files, for example `classes/db/MySQL.php`. Because base file `classed/db/db.php` has new abstract method, attempt to load mysql.php file results in 500 error code

The fix should be easy -- autoloader shouldn't do any discovery and load blindly. Instead, build script should populate some class index, and autoloader should always use this index.

**Comment by Traumflug** [comment]
You mean, autoloader picks up these classes when building the class index, right?

Having an index created at build time wouldn't work for shops running off a Git clone.

A better solution might be to let Core Updater remove surplus files in these directories. Currently, a merchant has to click all the checkboxes of such files to have them removed. For `classes/` and `controllers/`, it could not show these checkboxes, but always remove surplus files.

**Comment by getdatakick** [comment]
I agree that Core Updater should clean up any mess in these core directories. 

I'm not sure if I would consider it a solution to this problem, though. For example, some merchants might not use core updater at all, and do upgrade manually. So this fix might never be applied.

But it's probably good enough solution, and quite safe to do.

**Comment by getdatakick** [comment]
There are other code areas that can load php files, for example translation tools. These should use whitelist file as well.

https://forum.thirtybees.com/topic/4001-bug-in-translation-field-names/
